common/src/main/java/com/netflix/conductor/common/metadata/tasks/PollData.java

```diff
@@ -102,8 +107,7 @@ public synchronized boolean equals(Object obj) {
 		} else if (!queueName.equals(other.queueName))
 			return false;
 		if (workerId == null) {
-			if (other.workerId != null)
-				return false;
+			return other.workerId == null;
```

Comment:
consideration: I like Jave 7's  "if(!Objects.equals(x,y) return false" statements, a bit more concise, handles nulls, and leaves less room for error if somebody adds more checks at the end of the method for example and other.workerId == null returns true

Reply:
I agree, let's regenerate equals() and hashCode(), also remove synchronized keyword from these methods.